core/server/common/src/main/java/alluxio/master/audit/AsyncUserAccessAuditLogWriter.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.audit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * {@link AsyncUserAccessAuditLogWriter} writes user access audit log entries asynchronously.
+ */
+@ThreadSafe
+public final class AsyncUserAccessAuditLogWriter {
+  private static final int QUEUE_SIZE = 10000;
+  private static final Logger AUDIT_LOG =
+      LoggerFactory.getLogger(AsyncUserAccessAuditLogWriter.class);
```

Comment:
These two lines in log4j.properties:
log4j.category.alluxio.master.audit=INFO, ${alluxio.master.audit.logger.type}
log4j.additivity.alluxio.master.audit=false

Then this line in AsyncUserAccessAuditLogWriter:
rivate static final Logger AUDIT_LOG =
      LoggerFactory.getLogger(AsyncUserAccessAuditLogWriter.class);

The AsyncUserAccessAuditLogWriter is part of package alluxio.master.audit, therefore log4j is able to identify that the logger here is ${alluxio.master.audit.logger.type}